stlab/iterator/set_next.hpp

```diff
@@ -0,0 +1,62 @@
+/*
+    Copyright 2013 Adobe
+    Distributed under the Boost Software License, Version 1.0.
+    (See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+/*************************************************************************************************/
+
+#ifndef STLAB_ITERATOR_SET_NEXT_HPP
+#define STLAB_ITERATOR_SET_NEXT_HPP
+
+/*************************************************************************************************/
+
+namespace stlab {
+
+/*************************************************************************************************/
+
+namespace unsafe {
+
+/*************************************************************************************************/
+
+template <typename I> // I models NodeIterator
+struct set_next_fn;   // Must be specialized
```

Comment:
Not a big deal - but this pattern was a precursor to what Eric has done with customization points. http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/n4381.html We should probably move to Eric's model (which is a bit cleaner for the code doing the customization). If you copy the pattern from Eric's paper, note that symbols starting with `__` are reserved by the standard - drop the `__`.

Reply:
I'd like to defer those changes until after this forest work is checked in. It'll take me some time to digest his paper.